.github/workflows/ci.yml

```diff
@@ -43,8 +43,53 @@ jobs:
         run: |
           mypy --config-file mypy.ini --ignore-missing-imports --strict --warn-unreachable snuba
 
+  snuba-image:
+    name: Build snuba CI image
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    outputs:
+      branch: ${{ steps.branch.outputs.branch }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Get branch name (merge)
+        id: branch
+        run: echo "::set-output name=branch::$(echo ${GITHUB_REF#refs/heads/} | tr / -)"
+
+      # We are only using ghcr here for CI as `setup-gcloud` is a bit slow
+      # Should revisit this when we move off of google cloud build (we may want to move these to GCR)
+      - name: Registry login
+        run: |
+          echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
+
+      # These are pulled in order to be able to use docker layer caching
+      - name: Pull snuba CI images
+        run: |
+          docker pull ghcr.io/getsentry/snuba-ci:${{ github.sha }} || \
+            docker pull ghcr.io/getsentry/snuba-ci:${{ steps.branch.outputs.branch }} || \
+            docker pull ghcr.io/getsentry/snuba-ci:latest || true
+
+      - name: Build snuba docker image for CI
+        run: |
```

Comment:
I believe you need to add `--build-arg BUILDKIT_INLINE_CACHE=1` and add the env `DOCKER_BUILDKIT=1` in order to use `--cache-from`.

> To use an image as a cache source, cache metadata needs to be written into the image on creation. This can be done by setting --build-arg BUILDKIT_INLINE_CACHE=1 when building the image. After that, the built image can be used as a cache source for subsequent builds.

> This feature requires the BuildKit backend. You can either enable BuildKit or use the buildx plugin. The previous builder has limited support for reusing cache from pre-pulled images.

> To use the BuildKit backend, you need to set an environment variable DOCKER_BUILDKIT=1 on the CLI before invoking docker build.

Reply:
I can test this out more in a separate PR, but caching works as it is now: https://github.com/getsentry/snuba/pull/2017/checks?check_run_id=3126752853#step:4:24

I think `BUILDKIT_INLINE_CACHE` is so that we don't need to initially pull the images, but I wasn't able to get it to work.

Reply:
Here's a comparison of different combinations: https://github.com/getsentry/snuba/pull/2019/checks?check_run_id=3127238254

We can use Buildkit and get rid of the `docker pull` if we wanted.